### PR TITLE
fix: trigger jobs only when there is a GUID

### LINKED ISSUE #1420: Random duplicate jobs
Randomly, duplicate jx jobs are created a few seconds apart (2-3 seconds). 
This leads to wasting cloud resources.
Looking at the lighthouse webhook pod, only shows one lighthouse job was created, but build controller is creating 2 pipeline activities, which makes sense as there are 2 lighthouse jobs which creates 2 tekton pipelineruns.

I am not sure what is the root cause, but the second job does not have a GUID associated with it. It does have the `created-by-lighthouse` label set to `true`.
So, we can probably check if a GUID exists before creating a lighthousejob.

Long term: we want to understand the root cause of the duplicate jobs (a rogue goroutine may be)?

### PR DESCRIPTION
**Pull request by ankitm123**
Signed-off-by: ankitm123 <ankitmohapatra123@gmail.com>

fixes #1420

### CODE REVIEW

File: pkg/plugins/trigger/pull-request.go [hunk 1]
```diff
@@ -219,5 +220,8 @@ func buildAll(c Client, pr *scm.PullRequest, eventGUID string, elideSkippedConte
 	if err != nil {
 		return err
 	}
+	if eventGUID == "" && os.Getenv("GIT_KIND") == "github" {
```
Comment (ankitm123) [inline pkg/plugins/trigger/pull-request.go:223]: Not sure how other scm providers deal with GUID, so let's keep it only for github

**Comment by keskad** [comment]
What's the reason behind this?

**Comment by ankitm123** [comment]
> What's the reason behind this?

Check the linked issue: #1420 
It's weird that there are no entries in the lighthouse pod logs. But the duplicate job was created by lighthouse. I am running this in my cluster for a week now, haven't seen a duplicate job since that.

**Comment by keskad** [comment]
Damn, sorry I didn't notice the ticket. I see duplicated jobs on my test cluster, but I also see that github fires webhook twice for unknown for me reason.

**Comment by ankitm123** [comment]
> but I also see that github fires webhook twice for unknown for me reason.

For the same event? I dont see that issue though, but wont be surprised if sometimes github does that.

**Comment by keskad** [comment]
Yes. For the same event. Currently after bootjob passed and webhook was recreated I no longer see this issue. But previously I saw in github panel that github sent request twice in same second. Maybe that could be related somehow.

**Comment by ankitm123** [comment]
/close
This actually will breal manually triggered pipelines!

**Pull request closed without merging**